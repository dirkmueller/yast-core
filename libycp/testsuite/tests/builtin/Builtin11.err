Parsed:
----------------------------------------------------------------------
{
    // term MyTerm
    // term NewTerm
    // filename: "tests/builtin/Builtin11.ycp"
    term MyTerm = `Termified (`Austria, `Belgium);
    term NewTerm = `Termified ();
    return MyTerm;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // term MyTerm
    // term NewTerm
    // filename: "tests/builtin/Builtin11.ycp"
    term MyTerm = `Termified (`Austria, `Belgium);
    term NewTerm = `Termified ();
    NewTerm = add (MyTerm, `China);
    return (((((select (MyTerm, 0, nil) == `Austria) && (select (MyTerm, 1, nil) == `Belgium)) && (select (NewTerm, 0, nil) == `Austria)) && (select (NewTerm, 1, nil) == `Belgium)) && (select (NewTerm, size (MyTerm), nil) == `China));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // term MyTerm
    // term NewTerm
    // filename: "tests/builtin/Builtin11.ycp"
    term MyTerm = `Termified (`Austria, `Belgium);
    term NewTerm = `Termified ();
    MyTerm = `Termified ();
    NewTerm = add (MyTerm, `China);
    NewTerm = add (NewTerm, `China);
    return (((select (NewTerm, 0, nil) == `China) && (select (NewTerm, 1, nil) == `China)) && (symbolof (MyTerm) == `Termified));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/builtin/Builtin11.ycp"
    return "test case add for term finished";
}
----------------------------------------------------------------------
