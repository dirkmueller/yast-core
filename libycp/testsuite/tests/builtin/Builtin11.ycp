
# ---------------------------------------------------------
#
#  Filename:    Builtin11.ycp
#
#  Purpose:     test cases for YCP test
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------

// Creates a new term that is identical to the term  but has
// one more argument .
{
    term  MyTerm = `Termified( `Austria, `Belgium );
    term  NewTerm = `Termified( );

    return( MyTerm );
}

{
    term  MyTerm = `Termified( `Austria, `Belgium );
    term  NewTerm = `Termified( );

//  "one entry within the list is added"
    NewTerm = add( MyTerm, `China );
    
    return( ( (symbol)select( MyTerm, 0, nil ) == `Austria ) &&
            ( (symbol)select( MyTerm, 1, nil ) == `Belgium ) &&
            ( (symbol)select( NewTerm, 0, nil ) == `Austria ) &&
            ( (symbol)select( NewTerm, 1, nil ) == `Belgium ) &&
            ( (symbol)select( NewTerm, size( MyTerm ), nil ) == `China ) );
}

{
    term  MyTerm = `Termified( `Austria, `Belgium );
    term  NewTerm = `Termified( );

//  "one entry is added (append) to an empty term"
    MyTerm = `Termified( );
    NewTerm = add( MyTerm, `China );
    NewTerm = add( NewTerm, `China );
    return( 
            ( (symbol)select( NewTerm, 0, nil) == `China ) &&
            ( (symbol)select( NewTerm, 1, nil) == `China ) &&
            ( symbolof( MyTerm ) == `Termified )
          );
}

{
    return( "test case add for term finished" );
}
