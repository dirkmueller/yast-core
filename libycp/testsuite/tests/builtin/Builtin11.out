(`Termified (`Austria, `Belgium))
(true)
(true)
("test case add for term finished")
